packages/react-admin/src/mui/button/SaveButton.js

```diff
@@ -37,12 +38,22 @@ const sanitizeRestProps = ({
 
 export class SaveButton extends Component {
     handleClick = e => {
-        if (this.props.saving) {
+        const {
+            handleSubmitWithRedirect,
+            invalid,
+            redirect,
+            saving,
+            showNotification,
+        } = this.props;
+
+        if (saving) {
             // prevent double submission
             e.preventDefault();
         } else {
+            if (invalid) {
```

Comment:
I'm not sure what `invalid` means for redux-form, but can you confirm this notification shows only *after* the user has submitted the form?

Reply:
I confirm because we only check for an invalid form inside the click handler :)